### PR TITLE
Add encoding-type support for S3 ListObjects and more logging

### PR DESCRIPTION
**Pull request by lucyge2022**
### What changes are proposed in this pull request?

1. respect encoding-type flag in list-objects ( list-objects-v2 equivalent) and url-encoding the following return fields:
       Prefix, Key, and StartAfter
2. add util to print stacktrace of throwable in existing print thread stack trace util func
3. add service name in the webservice serving threads when starting for better tracing purpose
4. add access log in CompleteMultipartUploadHandler as well
5. add logging in CompleteMultipartUploadTask for better exception tracking.

### Why are the changes needed?

1. to fix the bug where  + in object names is shown as space when using aws s3 clit -> respect encoding-type flag fixed this
2. track completemultipartupload exception for better debugging purpose.

### Does this PR introduce any user facing changes?

N/A

### CODE REVIEW

File: core/server/proxy/src/main/java/alluxio/proxy/s3/CompleteMultipartUploadHandler.java [hunk 1]
```diff
@@ -161,15 +170,19 @@ public void handle(String s, Request request, HttpServletRequest httpServletRequ
...
-        LOG.error(e.toString());
+        LOG.error(ThreadUtils.formatStackTrace(cause));
       }
       httpServletResponse.getWriter().flush();
       request.setHandled(true);
     } catch (Exception e) {
       // This try-catch is not intended to handle any exceptions, it is purely
       // to ensure that encountered exceptions get logged.
-      LOG.error("Unhandled exception for {}. {}", s, e);
+      LOG.error("Unhandled exception for {}. {}", s, ThreadUtils.formatStackTrace(e));
```
Comment (tcrain) [inline core/server/proxy/src/main/java/alluxio/proxy/s3/CompleteMultipartUploadHandler.java:180]: I think if you want to print the stack trace you you just do something like this
LOG.error("Unhandled exception for {}.", s, e);

**Comment by tcrain** [comment]
I think you might need a 
`  @JsonInclude(JsonInclude.Include.NON_NULL)`
Annotation here, https://github.com/Alluxio/alluxio/blob/master/core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java#L364
 what do you think?

Also do you know if there are any tests checking the output, I couldn't see any? Maybe they should also include test checks for these additional fields.

**Comment by lucyge2022** [comment]
> I think you might need a ` @JsonInclude(JsonInclude.Include.NON_NULL)` Annotation here, https://github.com/Alluxio/alluxio/blob/master/core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java#L364 what do you think?
> 
> Also do you know if there are any tests checking the output, I couldn't see any? Maybe they should also include test checks for these additional fields.

As discussed offline, this annotation will make the field always be included in the output, if the field is null, it will be a tag with empty string, e.g. <tag></tag>, so with no annotation will align with aws s3 spec that if encoding-type is not given, it will not appear in the xml output response.

File: core/server/proxy/src/main/java/alluxio/proxy/s3/CompleteMultipartUploadHandler.java (hunk 1 above)
Comment (lucyge2022) [reply core/server/proxy/src/main/java/alluxio/proxy/s3/CompleteMultipartUploadHandler.java:180]: yeah i would like to do some other customized things like cap stack trace layer in the util func in the future, for now would like to print full thing for some debugging purpose.

**Comment by lucyge2022** [comment]
alluxio-bot, check this please

**Comment by Xenorith** [comment]
alluxio-bot, merge this please

**Comment by Xenorith** [comment]
alluxio-bot, cherry-pick this to branch-2.8 please

**Pull request merged**